src/commands/howto.ts

```diff
@@ -0,0 +1,193 @@
+- Please, do _NOT_  post direct links to kata trainers: make sure you don't have a trailling ~~\`/trainer\`~~ in the url.
+- Remember to also mention a title of the kata you are linking to.`,
+    reactions: [],
+  },
+  {
+    name: "create_thread",
+    description: "How to create discord threads",
+    body: `# How to create Discord threads
+Please use Discord threads to keep discussions focused, avoid spoilers, and to not introduce unnecessary noise into main channels. Remember to give a meaningful title to your thread, for example _"hobovsky's thread - help to solve Multiply kata"_, or something similar.
+
+See Discord docs if you do not know how to create threads: <https://support.discord.com/hc/en-us/articles/4403205878423-Threads-FAQ#h_01GDXVYE6AZA3QRSB42F5AGGYG>`,
+    reactions: [],
+  },
+];
+
+export const data = async () => {
+  const addTargetUserParam = (scb: SlashCommandSubcommandBuilder) =>
+    scb.addUserOption((b) =>
+      b.setName("user").setDescription("User to send the message to").setRequired(false)
+    );
+
+  let rootCommandBuilder = new SlashCommandBuilder()
+    .setName("howto")
+    .setDescription("HOWTOs and tutorials");
+
+  for (let subcommand of commands) {
+    rootCommandBuilder.addSubcommand((scb) =>
+      addTargetUserParam(scb.setName(subcommand.name).setDescription(subcommand.description))
+    );
+  }
+  return rootCommandBuilder.toJSON();
+};
+
+const allowedRoles = new Set(["admin", "mods", "mods+", "power-users"]);
+
+const hasSufficientPrivilege = (member: GuildMember | APIInteractionGuildMember | null) => {
+  let guildMember: GuildMember = member as GuildMember;
+  return guildMember && guildMember.roles.cache.some((role) => allowedRoles.has(role.name));
+};
+
+export const call = async (interaction: ChatInputCommandInteraction) => {
+  let invokingUser = interaction.user;
+  let targetUser = interaction.options.getUser("user", false) ?? invokingUser;
+
+  if (targetUser.bot) {
+    interaction.reply({
+      content: `${userMention(invokingUser.id)}, you cannot use this command on a bot.`,
+      ephemeral: true,
+    });
+    return;
+  }
+
+  let selfTarget = targetUser.id === invokingUser.id;
+  if (selfTarget || hasSufficientPrivilege(interaction.member)) {
+    let subCommand = interaction.options.getSubcommand();
+    let dmReply = commands.find((c) => c.name == subCommand);
+
+    if (dmReply) {
+      postHowtoDm(dmReply, targetUser).then(
+        () =>
+          interaction.reply({
+            content: `${userMention(targetUser.id)} please check your DMs`,
+            ephemeral: selfTarget,
+          }),
+        (reason: DiscordAPIError) =>
+          interaction.reply({
+            content: `Cannot send a DM to user ${targetUser.username}: ${reason.message}`,
```

Comment:
`/introduce` and `/warn` stores text under https://github.com/codewars/discord-bot/tree/main/text so we can show a link to the message when DM fails.

Reply:
I think I did similar thing now and a link is presented on failure, but the hardcoded URL is not perfect. Is it possible to move the repo/branch URL to some config?

Reply:
It's possible to make the repository URL configurable with something like this:

```typescript
let url = `https://github.com/${process.env.GITHUB_USER ?? "codewars"}/discord-bot/blob/main/text/howto/${subCommand}.md`;
```

But I doubt that's worth doing. Just confirm that the URL is correct.

Reply:
As far as I see, the URL used by me is fine.